### PR TITLE
Implemented hex_get_bytes function

### PR DESCRIPTION
**Pull request by zh-beep**
Hi @fhightower - thank you for the opportunity to contribute, I have added and tested this function. It returns a string represented in hex.

### CODE REVIEW

File: d8s_math/maths.py [hunk 1]
```diff
@@ -688,3 +688,15 @@ def number_to_engineering_notation(number):
 
     decimal_form = decimal.Decimal(number)
     return decimal_form.normalize().to_eng_string()
+
+def hex_get_bytes(hex_number,number_of_bytes):
```
Comment (fhightower) [inline d8s_math/maths.py]: Minor style improvement:

```suggestion
def hex_get_bytes(hex_number, number_of_bytes):
```

File: d8s_math/maths.py [hunk 2]
```diff
@@ -688,3 +688,15 @@ def number_to_engineering_notation(number):
...
+    length = len(hex(hex_number)) - 2
+    hex_num_bytes = length/2
+    if ((hex_num_bytes < number_of_bytes) or (hex_num_bytes == number_of_bytes)):
+        return '0'
+
```
Comment (fhightower) [inline d8s_math/maths.py]: I recommend removing the line between the `if` and `else`:

```suggestion
```

**Review by fhightower — CHANGES_REQUESTED** [review]
Thanks for working on this @zh-beep ! I do have a few questions.

I was testing your function with the hex value `0x12345678` and got the following results:

```python
hex_get_bytes(0x12345678, 1)  # '0x123456'
hex_get_bytes(0x12345678, 2)  # '0x1234'
hex_get_bytes(0x12345678, 3)  # '0x12'
hex_get_bytes(0x12345678, 4)  # '0'
```

Unfortunately, those results are not correct. Below is a table showing the value returned by the function in this PR and the correct values:

| Function Call | Actual Result  | Correct Result |
--- | --- | ---
`hex_get_bytes(0x12345678, 1)` | `'0x123456'` | `'0x12'`
`hex_get_bytes(0x12345678, 2)` | `'0x1234'` | `'0x1234'`
`hex_get_bytes(0x12345678, 3)` | `'0x12'` | `'0x123456'`
`hex_get_bytes(0x12345678, 4)` | `'0'` | `'0x12345678'`

Does that make sense?

As described in the "Byte" section [here](http://www.edu4java.com/en/concepts/hexadecimal-binary-number-byte-bit-word.html), each byte is represented in hexadecimal as two hexadecimal characters. So, `0x12345678` in hexadecimal represents four bytes (`12` represents one byte, `34` another byte, `56` another, and `78` another). So, if we want to get one byte of `0x12345678`, we expect to get `'0x12'`. Two bytes would give us `'0x1234'` and three would give us `'0x123456'`.

The actual output appears to be the reverse of what we want, so there should be a relatively easy change to make it work properly.

**Comment by zh-beep** [comment]
@fhightower - I updated the function - thank you for your help and for pointing out my mistake. I also updated the README to include the hex_get_bytes function.

I used [this package](https://github.com/gruns/icecream) to automate printing/debugging - thought it was pretty nifty: 

image

**Comment by fhightower** [comment]
Nice, looks good! I'll merge this in 🎉 .

Thanks for your contribution! If there are any other issues in the [Democritus project](https://github.com/democritus-project) that interest you, feel free to work on them!

**Pull request merged**